crates/turbopack-core/src/reference_type.rs

```diff
@@ -0,0 +1,123 @@
+use std::fmt::Display;
+
+// These enums list well known types, which we use internally. Plugins might add
+// custom types too.
+
+// TODO when plugins are supported, replace u8 with a trait that defines the
+// behavior.
+
+#[turbo_tasks::value(serialization = "auto_for_input")]
+#[derive(Debug, Clone, PartialOrd, Ord, Hash)]
+pub enum CommonJsReferenceSubType {
+    Custom(u8),
+    Undefined,
```

Comment:
Can we rename these `Undefined` subtypes to `Any`? Particularly because of the way it's used in the `ModuleRule` and `::includes`, an undefined subtype will match any subtype of the same `ReferenceType`

Reply:
Yeah we could. It's a bit weird since Undefined/Any is used on both side. You can use Undefined/Any in a reference to declare it as "sub type not defined" and you can use it as condition to say, you "accept any sub type".

But let's say that all asset references "should" set a sub type, so we can use Any for condition only.